### ISSUE TITLE
Visual Studio toolbox is not updated when upgrading the toolkit from v15.1 to v16.1

### ISSUE THREAD

**Issue by jmal327** [DevExpress/AjaxControlToolkit#224]
### The AJAX Control Toolkit \<VERSION\>
16.1.1
### The installation method
I am trying to use 16.1.1

### Minimal steps to reproduce the bug
Download and run AjaxControlToolkit.Installer.16.1.1.exe .
### Actual result
VSIX Installer error code: 2003 (VSIXInstaller.NoApplicableSKUsException)

2016-09-20 02:26:17 | INFO  | App.Application_Startup                  | Installer started 
2016-09-20 02:26:17 | INFO  |                                              VSIX name: ASP.NET AJAX Control Toolkit 
2016-09-20 02:26:17 | INFO  |                                              VSIX id: e79bade7-6755-466f-9788-3d8243bdcc5f 
2016-09-20 02:26:17 | INFO  |                                              VSIX version: 16.1.1.0 
2016-09-20 02:26:17 | INFO  | App.Application_Startup                  | Process Info 
2016-09-20 02:26:17 | INFO  |                                              OS: Microsoft Windows NT 6.1.7601 Service Pack 1 
2016-09-20 02:26:17 | INFO  |                                              Is 64bit: True 
2016-09-20 02:26:17 | INFO  |                                              Is admin: True 
2016-09-20 02:26:18 | INFO  | EnvironmentInfo.VsIsInstalled            | VisualStudio 2010 PROCore detected 
2016-09-20 02:26:18 | INFO  | VsUtility.GetPendingDeletions            | Pending deletions list for 2010 
2016-09-20 02:26:18 | INFO  |                                              e79bade7-6755-466f-9788-3d8243bdcc5f,15.1.4.0 C:\USERS\JEMALONE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\10.0\EXTENSIONS\SHPQVCIJ.4GL\ 
2016-09-20 02:26:18 | INFO  |                                              e79bade7-6755-466f-9788-3d8243bdcc5f,16.1.1.0 C:\USERS\JEMALONE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\10.0\EXTENSIONS\TJTXZHXB.XO0\ 
2016-09-20 02:26:18 | INFO  | EnvironmentInfo.GetInstallDirFromHklm    | InstallDir for 2010: c:\Program Files (x86)\Microsoft Visual Studio 10.0\Common7\IDE\ 
2016-09-20 02:26:18 | TRACE | EnvironmentInfo.GetInstallerPath         | c:\Program Files (x86)\Microsoft Visual Studio 10.0\Common7\IDE\VSIXInstaller.exe EXISTS 
2016-09-20 02:26:18 | INFO  | EnvironmentInfo.VsIsInstalled            | VisualStudio 2012 not detected 
2016-09-20 02:26:18 | INFO  | EnvironmentInfo.VsIsInstalled            | VisualStudio 2013 professional detected 
2016-09-20 02:26:18 | INFO  | VsUtility.GetPendingDeletions            | Pending deletions list for 2013 
2016-09-20 02:26:18 | INFO  |                                              e79bade7-6755-466f-9788-3d8243bdcc5f,15.1.4.0 C:\USERS\JEMALONE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\12.0\EXTENSIONS\ICRULKTA.PGF\ 
2016-09-20 02:26:18 | INFO  |                                              e79bade7-6755-466f-9788-3d8243bdcc5f,16.1.1.0 C:\USERS\JEMALONE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\12.0\EXTENSIONS\ONEN5O25.DBY\ 
2016-09-20 02:26:18 | INFO  | EnvironmentInfo.GetInstallDirFromHklm    | InstallDir for 2013: C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\IDE\ 
2016-09-20 02:26:18 | TRACE | EnvironmentInfo.GetInstallerPath         | C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\IDE\VSIXInstaller.exe EXISTS 
2016-09-20 02:26:18 | INFO  | EnvironmentInfo.VsIsInstalled            | VisualStudio 2015 not detected 
2016-09-20 02:26:18 | INFO  | EnvironmentInfo.GetEnvironmentInfo       | Vsix installer path is C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\IDE\VSIXInstaller.exe 
2016-09-20 02:26:18 | INFO  | MainWindow..ctor                         | Installed VS 
2016-09-20 02:26:18 | INFO  |                                              2010, ACT version: 16.1.1.0 C:\Users\jemalone\AppData\Local\Microsoft\VisualStudio\10.0\Extensions\tjtxzhxb.xo0 
2016-09-20 02:26:18 | INFO  |                                              2013, ACT version: 16.1.1.0 C:\Users\jemalone\AppData\Local\Microsoft\VisualStudio\12.0\Extensions\onen5o25.dby 
2016-09-20 02:26:18 | INFO  | Installer..ctor                          | User documents ACT directory C:\Users\jemalone\Documents\ASP.NET AJAX Control Toolkit 
2016-09-20 02:26:18 | TRACE | PageInstall..ctor                        | Initializing buttons 
2016-09-20 02:26:18 | INFO  | PageInstall.InitButtons                  | Buttons initialization info 
2016-09-20 02:26:18 | INFO  |                                              ACT extension is not installed in any supported Vs: False 
2016-09-20 02:26:18 | INFO  |                                              ACT extension is installed in all supported VSs: True 
2016-09-20 02:26:18 | INFO  |                                              ACT extension updated is required: False 
2016-09-20 02:26:18 | TRACE | PageInstall..ctor                        | Install page initialized 
2016-09-20 02:26:22 | TRACE | MainWindow.ShowUninstallProgressPage     | Progress page is shown 
2016-09-20 02:26:22 | INFO  | VsUtility.GetPendingDeletions            | Pending deletions list for 2010 
2016-09-20 02:26:22 | INFO  |                                              e79bade7-6755-466f-9788-3d8243bdcc5f,15.1.4.0 C:\USERS\JEMALONE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\10.0\EXTENSIONS\SHPQVCIJ.4GL\ 
2016-09-20 02:26:22 | INFO  |                                              e79bade7-6755-466f-9788-3d8243bdcc5f,16.1.1.0 C:\USERS\JEMALONE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\10.0\EXTENSIONS\TJTXZHXB.XO0\ 
2016-09-20 02:26:22 | INFO  | VsUtility.GetPendingDeletions            | Pending deletions list for 2013 
2016-09-20 02:26:22 | INFO  |                                              e79bade7-6755-466f-9788-3d8243bdcc5f,15.1.4.0 C:\USERS\JEMALONE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\12.0\EXTENSIONS\ICRULKTA.PGF\ 
2016-09-20 02:26:22 | INFO  |                                              e79bade7-6755-466f-9788-3d8243bdcc5f,16.1.1.0 C:\USERS\JEMALONE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\12.0\EXTENSIONS\ONEN5O25.DBY\ 
2016-09-20 02:26:22 | INFO  | VsixInstallerWrapper.Run                 | Run VSIXInstaller with arguments: '/quiet /uninstall:e79bade7-6755-466f-9788-3d8243bdcc5f' 
2016-09-20 02:26:32 | INFO  | VsixInstallerWrapper.Run                 | VSIXInstaller exit code: 2003 
2016-09-20 02:26:32 | INFO  | Installer.Uninstall                      | VSIXInstaller log:

9/20/2016 2:26:23 PM - Microsoft VSIX Installer
9/20/2016 2:26:23 PM - -------------------------------------------
9/20/2016 2:26:23 PM - Initializing Uninstall...
9/20/2016 2:26:32 PM - VSIXInstaller.NoApplicableSKUsException: The extension with ID 'e79bade7-6755-466f-9788-3d8243bdcc5f' is not installed to any currently installed products.
   at VSIXInstaller.App.InitializeUninstall()
   at VSIXInstaller.App.OnStartup(StartupEventArgs e)
 
2016-09-20 02:26:32 | TRACE | Installer.TryRemoveDirectory             | Removing directory: C:\Users\jemalone\AppData\Local\Microsoft\VisualStudio\10.0\Extensions\ASP.NET AJAX Control Toolkit 
2016-09-20 02:26:32 | WARN  | Installer.TryRemoveDirectory             | C:\Users\jemalone\AppData\Local\Microsoft\VisualStudio\10.0\Extensions\ASP.NET AJAX Control Toolkit{0} not found 
2016-09-20 02:26:32 | TRACE | Installer.TryRemoveDirectory             | Removing directory: C:\Users\jemalone\AppData\Local\Microsoft\VisualStudio\12.0\Extensions\ASP.NET AJAX Control Toolkit 
2016-09-20 02:26:32 | WARN  | Installer.TryRemoveDirectory             | C:\Users\jemalone\AppData\Local\Microsoft\VisualStudio\12.0\Extensions\ASP.NET AJAX Control Toolkit{0} not found 

### Expected result
I expect the toolkit to install
### Browser(s) used
IE
### A site deployment method
- [ ] VS development web-server,
- [X ] IIS

**Comment by jmal327** [comment]
The installer fails if it doesn't find the directory.  I wish it would just install, I really don't care about an uninstall first.

**Comment by MikhailTymchukDX** [comment]
Hi @jmal327 ,
Thank you for the detailed report of this issue.

The log shows that you already have v16.1.1 installed, so the installer tries to uninstall the latest version by your request, but fails for some reason.

Would you please provide a VSIX installer log located at `%TEMP%\VSIXInstaller_[GUID_HERE]` path?

**Comment by jmal327** [comment]
Good morning Mikhail.  What happened was I ended up uninstalling it via the Visual Studio Tools>Extensions and Updates.  Then, after that, when I re-run the installer I got the option to repair the installation.  I did that and now the project runs fine, but I get an error every time I start Visual Studio.

Now, when I start Visual Studio I get:

![roamingerror](https://cloud.githubusercontent.com/assets/22328860/18711180/b06016fe-7fd6-11e6-835c-8466406258de.PNG)

I attached all versions (including a previous install of 15.1.1) of the VSIXInstaller logs here...

[VSIXInstaller_945a8c68-5b76-497e-a719-3e87ccd8ea9d.zip](https://github.com/DevExpress/AjaxControlToolkit/files/484975/VSIXInstaller_945a8c68-5b76-497e-a719-3e87ccd8ea9d.zip)

**Comment by MikhailTymchukDX** [comment]
I'm happy to hear that the toolkit operates correctly now.

It's hard to give precise advice without examining the `ActivityLog.xml`, but you can try to fix the package loading failure by clearing the component cache of Visual Studio: http://stackoverflow.com/a/23646013/644496

**Comment by jmal327** [comment]
Thanks!  I tried your suggestion and all is well!  Well actually I ran the devenv.exe /log and found out there was another visual studio extension that was causing the problem.  All is well now!

But I want you to realize my original situation, just so that you know.  I was on 15.1.1 and had problems with the z-index turning negative after 7 repeated open and close events.  I read online that version 16.1.1 would fix the problem.  So my problems started when I tried to upgrade in place from 15.1.1 to 16.1.1.   I ran the AjaxControlToolkit.Installer.16.1.1.exe.  Perhaps that was my first mistake.  After running AjaxControlToolkit.Installer.16.1.1.exe  I expected an automatic upgrade.  But the toolbox was still referencing 15.1.1.  So I ran AjaxControlToolkit.Installer.16.1.1.exe a second time.  Then all it offered was an uninstall.  I tried that, and the installer keep dropping and left me hanging.  That is when I first started this thread.
